components/builder-worker/src/server.rs

```diff
@@ -152,7 +152,12 @@ impl Server {
             let reply = self.runner_cli.recv_ack()?;
             self.fe_sock.send(reply, 0)?;
         }
-        Ok(())
+
+        // Currently, there's no good way to cancel a studio build that is in progress.
+        // Terminate the worker process (it will get restarted by the supervisor) in
+        // order to handle the case where a build is hung.
+        warn!("Exiting worker (cancel received)");
+        ::std::process::exit(0);
```

Comment:
If the build is hung, and we're effectively killing the worker, should we exit with a non-zero exit code?

Reply:
Probably fine to do that too.. I don't think it really matters here since we're not using the worker exit codes for anything.  One thing to clarify though - all job cancel requests will now exit the worker, not just hung builds.